DM-26447: Add support for options to the add command for the script commander

r-owen opened the pull request.

tribeiro commented on python/lsst/ts/scriptqueue/ui/script_queue_commander.py:
```diff
@@ -127,15 +154,60 @@ async def do_add(self, args):
+        config_start_ind = 2
+        options_start_ind = 2
+        if len(args) > 2 and args[2].startswith("@"):
+            config_path = args[2]
```
I think you need to strip the "`@`" of the name, otherwise it won't find the file unless it also has the "@" on it. 

I tried...

r-owen replied:
Good catch. I fixed that and also add a default suffix of .yaml (and documented that in the help string). Thank you.

tribeiro commented on python/lsst/ts/scriptqueue/ui/script_queue_commander.py:
```diff
@@ -31,17 +32,43 @@
+      -locationSalIndex=int
+      -logLevel=int  # error=40, warning=30, info=20, debug=10; default is {script_log_level}
+      -pauseCheckpoint=str  # a regex
+      -stopCheckpoint=str  # a regex"""
```
I think it would be helpful if you could add a couple of examples of an add command setting some values.. something like:

```
Example:

add s some_std_script.py @str_script_config_1.yaml -logLevel=10

add s some_std_script.py value1=10 value2=string_value -logLevel=10
```

r-owen replied:
Good suggestion. I did that.

tribeiro commented:
I also tried setting the configuration using the `keyword1=value1` and I don't think it is working.

tribeiro commented:
Another request, can you add a tab or maybe a `<<<<<<` at the beginning of the lines with prints from events etc? it is really hard to find things the way it is now, e.g.;

```
1598505621.784: script: cmdId=135838735, salIndex=100000, isStandard=True, path=maintel/enable_mtcs.py, timestampProcessStart=1598505436.6645, timestampConfigureStart=1598505462.9804, timestampConfigureEnd=1598505462.9890, timestampRunStart=1598505462.9951, timestampProcessEnd=1598505584.7838, processState=4, scriptState=9
1598505621.785 queue enabled=True, running=True, currentSalIndex=0, salIndices=[], pastSalIndices=[100000]
Finished command stopScripts
add s maintel/enable_mtcs.py mtm1m3="Default" -logLevel=10
1598505670.930 queue enabled=True, running=True, currentSalIndex=0, salIndices=[100001], pastSalIndices=[100000]
1598505670.944: script: cmdId=135838736, salIndex=100001, isStandard=True, path=maintel/enable_mtcs.py, timestampProcessStart=1598505633.9432, timestampConfigureStart=0.0000, timestampConfigureEnd=0.0000, timestampRunStart=0.0000, timestampProcessEnd=0.0000, processState=1, scriptState=0
Finished command add
```

could be

```
<<<<<< 1598505621.784: script: cmdId=135838735, salIndex=100000, isStandard=True, path=maintel/enable_mtcs.py, timestampProcessStart=1598505436.6645, timestampConfigureStart=1598505462.9804, timestampConfigureEnd=1598505462.9890, timestampRunStart=1598505462.9951, timestampProcessEnd=1598505584.7838, processState=4, scriptState=9
<<<<<< 1598505621.785 queue enabled=True, running=True, currentSalIndex=0, salIndices=[], pastSalIndices=[100000]
<<<<<< Finished command stopScripts
add s maintel/enable_mtcs.py mtm1m3="Default" -logLevel=10
<<<<<< 1598505670.930 queue enabled=True, running=True, currentSalIndex=0, salIndices=[100001], pastSalIndices=[100000]
<<<<<< 1598505670.944: script: cmdId=135838736, salIndex=100001, isStandard=True, path=maintel/enable_mtcs.py, timestampProcessStart=1598505633.9432, timestampConfigureStart=0.0000, timestampConfigureEnd=0.0000, timestampRunStart=0.0000, timestampProcessEnd=0.0000, processState=1, scriptState=0
<<<<<< Finished command add
```

r-owen commented:
> I also tried setting the configuration using the `keyword1=value1` and I don't think it is working.

You are right. I accidentally left some old code in when I refactored. I fixed that and tested the fix.

r-owen commented:
Regarding changing the output format: I really do not want to add more stuff at the beginning of each output line. The timestamp is intended to serve as a "this is a SAL message" indicator. I don't like it much, but of course having a timestamp is sometimes important. Perhaps I should use normal date format instead. It is easier to work with in some ways, but a real nuisance for computing elapsed time between two messages.

In any case, such changes would go into ts_salobj and I consider that out of scope for this ticket.

r-owen commented:
That is a great suggestion (though the need to add a --no-color option is a headache). Feel free to file a pull request for ts_salobj.

r-owen commented:
The biggest change I'd like to see is to add readline support so we have command recall and editing. And again that'd be in ts_salobj.
